app/models/custom_button_event.rb

```diff
@@ -1,2 +1,12 @@
 class CustomButtonEvent < EventStream
+  virtual_column :button_name, :type => :string
+  virtual_column :automate_entry_point, :type => :string
+
+  def automate_entry_point
+    full_data[:automate_entry_point]
```

Comment:
Based on the virtual column I would expect this to always return a string, so I would suggest adding a `to_s` incase it comes up nil.  My guess is if the UI allows sorting that would cause it to blow us.

cc @h-kataria

Reply:
@gmcculloug just tried with userid as `nil`, seems working

Reply:
@lfu Can you let me know what you are thinking here.  I'm suggesting that virtual columns should return their defined data type, which means we should ensure these methods backing the virtual columns return strings.

Reply:
@gmcculloug I agree with what you said. 👍